client/helper_test.go

```diff
@@ -0,0 +1,207 @@
+	`},` +
+	`{` +
+	`"kty": "RSA",` +
+	`"use": "sig",` +
+	`"kid": "igun4zI3Hs-O4vqncDpnoE5Ysn-ayBePCZ89dKqxdLA",` +
+	`"e": "AQAB",` +
+	`"n": "8kaNkiwPhmT2v8YVcoSJ72bG_9kJwvtStGUxz6RVRMEzXXZG70igKSI3-V9LkBoVdPeN8CLX3adv1vX5R18EhNUUFn88_RpdKOi0DKPbq1Rs_-obWtabtbGhMy7uo5BMVdmfBXjDmDJU11W44_FOjW3d21KjsozUR2LJkENfWS8A8yhbeGVzl1fv062DO86Dwfy5Zknph7irs3ioPc4PBrMhX55TuEE77841Xx4qIJKfKjFQy9--XNAm3qOJyMCQG_eL_tcCcU0Okl_5tz6PSEAzQ5G3WYA4DgHsRrk9FlvI7LyGVtObH70aPX3YK0Ggh6ieQFnomj6wcN6II6dgslgMr7a4fNblS0j3_k4twbhn-LFb39cFKZekPrxT4LO58kTtNxT4tnPOQxdjy96p4ES6YBYPerpIUMHp1nyL3yY_Uxub_gGKRq4fxLK3QrJBFigwg7isoglxd1lGZaaVVQwz5LxjeBut2sFkIrWTzae_hFZHI9D8v_WwzhbwmIJzfP64aqBV-Rloj83e4BKXItBqwBQrcJtAvo1IYLekBC6HON2vczECEPLOyx47BOry1M4kxZfu342Ya_8gSqRxKQKVuKh88aNHFTUqn4jn2MFErDdjXdlqjtZGti1vWqHAz6hQ2ZDoUEqjFCSKWJ5QmHMdc3AMJpKLNkuQWIFJcm8"` +
+	`}` +
+	`]` +
+	`}`
+
+type TestStorage struct {
+	data map[string]string
+}
+
+func (storage *TestStorage) GetItem(key string) string {
+	return storage.data[key]
+}
+
+func (storage *TestStorage) SetItem(key, value string) {
+	storage.data[key] = value
+}
+
+func TestFetchOidcConfigShouldReturnExpectedOidcConfig(t *testing.T) {
+	wantedOidcConfig := core.OidcConfigResponse{
+		AuthorizationEndpoint: "testAuthorizationEndpoint",
+	}
+
+	patches := gomonkey.ApplyFunc(core.FetchOidcConfig, func(client *http.Client, endpoint string) (core.OidcConfigResponse, error) {
+		return wantedOidcConfig, nil
+	})
+	defer patches.Reset()
+
+	logtClient := NewLogtoClient(&LogtoConfig{
+		Endpoint: "https://example.com",
+	}, &TestStorage{})
+
+	gotOidcConfig, fetchOidcConfigError := logtClient.fetchOidcConfig()
+
+	if fetchOidcConfigError != nil {
+		t.Fatal(fetchOidcConfigError)
+	}
+
+	if !cmp.Equal(gotOidcConfig, wantedOidcConfig) {
+		t.Fatalf("Expected Oidc Config: %v\nActual Oidc Config : %v", wantedOidcConfig, gotOidcConfig)
+	}
+}
+
+func TestLoadAccessTokenMapShouldLoadAccessTokenMapFromStorage(t *testing.T) {
+	expiresAt := time.Now().Unix() + 60
+
+	logtoConfig := &LogtoConfig{
+		Resources: []string{"testResource"},
+	}
+
+	testStorage := &TestStorage{
+		data: map[string]string{
+			StorageKeyIdToken:        "ThisIdTokenMakeIsAuthenticatedReturnTrue",
+			StorageKeyAccessTokenMap: `{"@testResource":{"token":"token","scope":"scope","expiresAt":` + strconv.FormatInt(expiresAt, 10) + `}}`,
+		},
+	}
+
+	wantedAccessToken := AccessToken{
+		Token:     "token",
+		Scope:     "scope",
+		ExpiresAt: expiresAt,
+	}
+
+	logtoClient := NewLogtoClient(logtoConfig, testStorage)
+
+	logtoClient.loadAccessTokenMap()
+
+	got := logtoClient.accessTokenMap["@testResource"]
```

Comment:
What about with scopes?

Reply:
We don't support scopes in V1, so the key will not contain the scope part.